src/main/java/de/muenchen/dave/services/messstelle/auswertung/SpreadsheetService.java

```diff
@@ -0,0 +1,290 @@
+package de.muenchen.dave.services.messstelle.auswertung;
+
+import de.muenchen.dave.domain.dtos.messstelle.FahrzeugOptionsDTO;
+import de.muenchen.dave.domain.dtos.messstelle.auswertung.Auswertung;
+import de.muenchen.dave.domain.dtos.messstelle.auswertung.AuswertungProMessstelle;
+import de.muenchen.dave.domain.dtos.messstelle.auswertung.MessstelleAuswertungIdDTO;
+import de.muenchen.dave.domain.dtos.messstelle.auswertung.MessstelleAuswertungOptionsDTO;
+import de.muenchen.dave.domain.elasticsearch.detektor.Messstelle;
+import de.muenchen.dave.domain.enums.AuswertungsZeitraum;
+import de.muenchen.dave.services.messstelle.MessstelleService;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.collections4.CollectionUtils;
+import org.apache.commons.collections4.ListUtils;
+import org.apache.commons.lang3.ObjectUtils;
+import org.apache.commons.lang3.StringUtils;
+import org.apache.poi.ss.usermodel.Cell;
+import org.apache.poi.ss.usermodel.CellType;
+import org.apache.poi.ss.usermodel.Row;
+import org.apache.poi.ss.usermodel.Sheet;
+import org.apache.poi.ss.usermodel.Workbook;
+import org.apache.poi.xssf.usermodel.XSSFWorkbook;
+import org.springframework.stereotype.Service;
+
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.math.BigDecimal;
+import java.util.ArrayList;
+import java.util.Comparator;
+import java.util.List;
+import java.util.Optional;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.concurrent.atomic.AtomicReference;
+
+@RequiredArgsConstructor
+@Service
+@Slf4j
+public class SpreadsheetService {
+
+    private final MessstelleService messstelleService;
+
+    public byte[] createFile(final List<AuswertungProMessstelle> auswertungenProMessstelle, final MessstelleAuswertungOptionsDTO options)
+            throws IOException {
+        final var spreadsheetDocument = new XSSFWorkbook();
```

Comment:
Wäre hier nicht das `Open Document Format for Office Applications (ODF)` besser, da es sich bei Dave um OSS handelt.

Reply:
Können wir mal mit Martin reden. Würde ich jetzt erstmal so lassen. XLSX kann ja auch von Open/LibreOffice bearbeitet werden